network/netplay/netplay_discovery.c

```diff
@@ -145,20 +155,27 @@ bool netplay_discovery_driver_ctl(enum rarch_netplay_discovery_ctl_state state,
 #if defined(SOL_SOCKET) && defined(SO_BROADCAST)
          if (setsockopt(lan_ad_client_fd, SOL_SOCKET, SO_BROADCAST,
                   (const char *)&canBroadcast, sizeof(canBroadcast)) < 0)
-             RARCH_WARN("Failed to set netplay discovery port to broadcast.\n");
+             RARCH_WARN("[discovery] Failed to set netplay discovery port to broadcast.\n");
 #endif
 
          /* Put together the request */
          memcpy((void *) &ad_packet_buffer, "RANQ", 4);
          ad_packet_buffer.protocol_version = htonl(NETPLAY_PROTOCOL_VERSION);
 
-         /* And send it off */
-         if (sendto(lan_ad_client_fd, (const char *) &ad_packet_buffer,
-            2*sizeof(uint32_t), 0, addr->ai_addr, addr->ai_addrlen) <
-            (ssize_t) (2*sizeof(uint32_t)))
-            RARCH_WARN("Failed to send netplay discovery response.\n");
+         for (int k = 0; k < interfaces.size; k++)
+         {
+            strlcpy(ad_packet_buffer.address, interfaces.entries[k].host,
+               NETPLAY_HOST_STR_LEN);
+            /* And send it off */
+            if (sendto(lan_ad_client_fd, (const char *) &ad_packet_buffer,
+               2*sizeof(uint32_t) + NETPLAY_HOST_STR_LEN, 0, addr->ai_addr, addr->ai_addrlen) <
```

Comment:
Why this length?

Reply:
this code was there already (except it was 2 * sizeof(uin32_t) only. Since I added an IP address to the mix I added NETPLAY_HOST_STR_LEN

I guess I should change it to sizeof(ad_packet_buffer)?